src/selectors/selectExplorerTree.ts

```diff
@@ -0,0 +1,223 @@
+import platformPath from 'path';
+import { Uri } from 'vscode';
+import { CaseHash } from '../cases/types';
+import { caseAdapter, jobAdapter, State } from '../data/slice';
+import { Job, JobHash, JobKind, PersistedJob } from '../jobs/types';
+import { LeftRightHashSetManager } from '../leftRightHashes/leftRightHashSetManager';
+import { buildHash, isNeitherNullNorUndefined } from '../utilities';
+
+export const doesJobAddNewFile = (kind: Job['kind']): boolean => {
+	return [
+		JobKind.copyFile,
+		JobKind.createFile,
+		JobKind.moveAndRewriteFile,
+		JobKind.moveFile,
+	].includes(kind);
+};
+
+export type ExplorerNodeHashDigest = string & {
+	__ExplorerNodeHashDigest: 'ExplorerNodeHashDigest';
+};
+
+export type ExplorerNode =
+	| Readonly<{
+			hashDigest: ExplorerNodeHashDigest;
+			kind: 'ROOT';
+			label: string;
+	  }>
+	| Readonly<{
+			hashDigest: ExplorerNodeHashDigest;
+			kind: 'TOP';
+			label: string;
+	  }>
+	| Readonly<{
+			hashDigest: ExplorerNodeHashDigest;
+			kind: 'DIRECTORY';
+			label: string;
+	  }>
+	| Readonly<{
+			hashDigest: ExplorerNodeHashDigest;
+			kind: 'FILE';
+			label: string;
+			jobHash: string;
+			fileAdded: boolean;
```

Comment:
do we need `fileAdded` and `jobHash` fields directly on the Node object? can we somehow make  ExplorerNode  objects to have the same shape? otherwise it can impact performance.

Reply:
the webview needs to know when to render a file added icon or a normal icon based on the job kind (which we calculate as `fileAdded`)